### DISCUSSION TITLE
A process hangs on Ubuntu sometimes

### DISCUSSION THREAD

**Discussion by peaceiris** [actions/runner#979] [Q&A] [answered]
The problem is that Hugo hangs on `runs-on: ubuntu-18.04` and `20.04` sometimes.

[Hanging at `hugo` step sometimes · Issue #465 · peaceiris/actions-hugo](https://github.com/peaceiris/actions-hugo/issues/465)

I tested some settings as follows:

- The building works well on local Ubuntu 18.04/20.04
- `runs-on: macos-latest` works well
- `container.image: 'ubuntu:18.04'` also hangs
- When I removed images, CSS, JavaScript, and HTML files from the user's content directory, then built it some times with a small set. It works well.

My questions:

- Does the Ubuntu VM of GitHub Actions runner has any limitation?
- Have anybody faced a problem like this with other tools or programs?
- Any ideas what I can try next to debug this?

Sorry for my ambiguous question.

**Comment by EnricoMi** [comment]
The VMs have only 2 CPUs / 4 vCPUs and around 7GB RAM. I managed to consistently hang them with `make -j`, which spawns unlimited number of parallel compile processes. This eventually brought the VM to an halt, without GitHub actions being able to cancel the job or retrieve logs.

**Reply by EnricoMi** [reply]
I just found out by running this command before the one that eventually hangs:

    bash -c "while true; do pstree -uA; sleep 1; done &"

This requires `psmisc` to be installed (`ubuntu-latest`). It continuously shows all existing processes, every second, while running the subsequent commands.

**Comment by peaceiris** [comment, marked as answer]
In our case, the problem has been related to Hugo.

> Closed by [Release v0.91.2 · gohugoio/hugo](https://github.com/gohugoio/hugo/releases/tag/v0.91.2)
> https://github.com/peaceiris/actions-hugo/issues/465#issuecomment-1000579347
